components/Datepicker/src/index.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import clsx from 'clsx';
+import {
+  format,
+  Locale,
+  getWeeksInMonth,
+  formatISO,
+  setDay,
+  addWeeks,
+  subMonths,
+  addMonths,
+  isSameMonth,
+  setDate,
+} from 'date-fns';
+import { nl } from './locale';
+
+import BaseProps from '@gemeente-denhaag/baseprops';
+
+import './datepicker.css';
+
+export interface DatepickerProps extends BaseProps {
+  /**
+   * Whether the input field is in a error state.
+   */
+  error?: boolean;
+  /**
+   * Whether the input field is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Locale (from @gemeente-denhaag/datepicker/locale or date-fns/locale) (default: nl)
+   */
+  locale?: Locale;
+  /**
+   * Starting date for calendar (default: today)
+   */
+  startDate?: Date;
+}
+
+interface DatepickerState {
+  current: Date;
+  selected?: Date;
+  opened: boolean;
+}
+
+/**
+ * The DatePicker component allows the user to select a date.
+ */
+export const Datepicker: React.FC<DatepickerProps> = ({
+  startDate = new Date(),
+  locale = nl,
+  ...props
+}: DatepickerProps) => {
+  const rootStyles = clsx(
+    'denhaag-datepicker',
+    {
+      'denhaag-datepicker--error': props.error,
+      'denhaag-datepicker--disabled': props.disabled,
+    },
+    props.className,
+  );
+
+  const [state, setState] = React.useState<DatepickerState>({
+    current: setDate(startDate, 1),
+    selected: undefined,
+    opened: false,
+  });
+
+  const componentRef = React.useRef<HTMLDivElement>(null);
+
+  const outsideClickListener = (e: Event) => {
+    if (!componentRef.current?.contains(e.target as Node)) {
+      setState({
+        ...state,
+        opened: false,
+      });
+    }
+  };
+
+  React.useEffect(() => {
+    document.addEventListener('click', outsideClickListener);
+    return () => {
```

Comment:
I prefer returning a named function as it's much more obvious what's happening when it's called `cleanup`, but that's a preference thing. (Caused by the fact that I had no idea what returning a function in `useEffect` would do)

Reply:
Fair enough, nice improvement to the readability. Now returns a named `outsideClickListenerCleanup` function.